<?xml version="1.0"?>
<launch>

  <!-- general arguments -->
  <arg name="spawn_window" default="True"/>

  <!-- Default argments -->
  <arg name="gnome_term_cmd" value="gnome-terminal --command"/>
  <arg name="xterm_cmd" value="xterm -e"/>
  <arg name="terminal_cmd" value="$(arg xterm_cmd)" if="$(arg spawn_window)"/>
  <arg name="terminal_cmd" value="" unless="$(arg spawn_window)"/>

  <!-- Robot Trajectory Execution -->
  <node name="robot_execution" type="robot_execution.py" pkg="gilbreth_grasp_planning" launch-prefix="$(arg terminal_cmd)" output="screen"/>

  <!-- Uncomment to see trajectory execution aborted caused by goal tolerence violation>
  <node name="robot_execution" type="robot_execution_goal_abortion.py" pkg="gilbreth_grasp_planning" launch-prefix="$(arg terminal_cmd)" output="screen"/-->

  <!-- Tool Pose Planning -->
  <rosparam command="load" file="$(find gilbreth_grasp_planning)/config/tool_plan_param.yaml" ns="gilbreth/tool_plan"/>    
  <node pkg="gilbreth_grasp_planning" name="tool_planning_node" type="tool_planner.py" launch-prefix="$(arg terminal_cmd)" output="screen"/>

</launch>
